app/components/dashboard/hypertension/patients_protected_component.rb

```diff
@@ -1,16 +1,15 @@
 class Dashboard::Hypertension::PatientsProtectedComponent < ApplicationComponent
-  attr_reader :data, :contactable, :period, :period_start
+  attr_reader :period, :period_start
 
-  def initialize(region:, data:, period:)
+  def initialize(region:, period:)
     @region = region
-    @data = data
     # Show only data up to last month
     @period = period.advance(months: -1)
     @period_start = period.advance(months: -36)
```

Comment:
I couldn't think of a cleaner way to make this work. The repository is already initialized in the region controller, however, the `MAX_MONTHS_OF_DATA` is configured for all the charts. You can't get a single chart to have a different number of months to display without affecting other charts. Or you'll have to make the repository much more composable (to accept different reporting periods for different columns), which is not something we can dedicate our time to at the moment.

Reply:
Acknowledge